core-web/libs/block-editor/src/lib/extensions/ai-content-actions/plugins/ai-content-actions.plugin.ts

```diff
@@ -0,0 +1,190 @@
+import { EditorView } from 'prosemirror-view';
+import { Subject } from 'rxjs';
+import tippy, { Instance, Props } from 'tippy.js';
+
+import { ComponentRef } from '@angular/core';
+
+import { takeUntil } from 'rxjs/operators';
+
+import { Editor } from '@tiptap/core';
+
+import { ACTIONS, AIContentActionsComponent } from '../ai-content-actions.component';
+import { AI_CONTENT_ACTIONS_PLUGIN_KEY } from '../ai-content-actions.extension';
+import { TIPPY_OPTIONS } from '../utils';
+
+interface AIContentActionsProps {
+    pluginKey: PluginKey;
+    editor: Editor;
+    element: HTMLElement;
+    tippyOptions?: Partial<Props>;
+    component: ComponentRef<AIContentActionsComponent>;
+}
+
+interface PluginState {
+    open: boolean;
+}
+
+export type AIContentActionsViewProps = AIContentActionsProps & {
+    view: EditorView;
+};
+
+export class AIContentActionsView {
+    public editor: Editor;
+
+    public node: Node;
+
+    public element: HTMLElement;
+
+    public view: EditorView;
+
+    public tippy: Instance | undefined;
+
+    public tippyOptions?: Partial<Props>;
+
+    public pluginKey: PluginKey;
+
+    public component: ComponentRef<AIContentActionsComponent>;
+
+    private destroy$ = new Subject<boolean>();
+
+    constructor(props: AIContentActionsViewProps) {
+        const { editor, element, view, tippyOptions = {}, pluginKey, component } = props;
+
+        this.editor = editor;
+        this.element = element;
+        this.view = view;
+
+        this.tippyOptions = tippyOptions;
+
+        this.element.remove();
+        this.pluginKey = pluginKey;
+        this.component = component;
+
+        this.component.instance.actionEmitter.pipe(takeUntil(this.destroy$)).subscribe((action) => {
+            switch (action) {
+                case ACTIONS.ACCEPT:
+                    this.acceptContent();
+                    break;
+
+                case ACTIONS.REGENERATE:
+                    this.generateContent();
+                    break;
+
+                case ACTIONS.DELETE:
+                    this.deleteContent();
+                    break;
+            }
+        });
+
+        this.view.dom.addEventListener('keydown', this.handleKeyDown.bind(this));
+    }
+
+    private acceptContent() {
+        this.editor.commands.closeAIContentActions();
+        const content = this.component.instance.getLatestContent();
+        this.editor.commands.insertContent(content);
+    }
+
+    private generateContent() {
+        this.editor.commands.closeAIContentActions();
+
+        this.component.instance
+            .getNewContent()
+            .pipe(takeUntil(this.destroy$))
+            .subscribe((newContent) => {
+                if (newContent) {
+                    this.editor.commands.deleteSelection();
+                    this.editor.commands.insertAINode(newContent);
+                    this.editor.commands.openAIContentActions();
+                }
+            });
```

Comment:
Every time we call this function, we are creating a new subscription. We can handle this in two different ways.

1. Moves this subscription to the `constructor` (Best option - IMO)
2. Use `take(1)` Pipe

```suggestion
        this.component.instance
            .getNewContent()
            .pipe(
              take(1), // Take only the first emission 
              takeUntil(this.destroy$) // If the component is destroyed first
            )
            .subscribe((newContent) => {
                if (newContent) {
                    this.editor.commands.deleteSelection();
                    this.editor.commands.insertAINode(newContent);
                    this.editor.commands.openAIContentActions();
                }
            });
```